libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2747,12 +2777,13 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		templ("gas", "sub(gas(), " + formatNumber(gasNeededByCaller) + ")");
 	}
 	// Order is important here, STATICCALL might overlap with DELEGATECALL.
+	std::string const callPrefix = eof ? "ext" : "";
 	if (isDelegateCall)
-		templ("call", "delegatecall");
+		templ("call", callPrefix + "delegatecall");
```

Comment:
I'd suggest doing it the simple way. It's not really longer and has the advantage that it will actually show up when grepping the code for `extdelegatecall`:

```suggestion
		templ("call", eof ? "extdelegatecall" : "delegatecall");
```

Same for `extcall` and `extstaticcall`. And also in `appendBareCall()` below.

Reply:
changed